lib/active_shipping/location.rb

```diff
@@ -42,48 +60,50 @@ def initialize(options = {})
     end
 
     def self.from(object, options = {})
-      return object if object.is_a? ActiveShipping::Location
-      attr_mappings = {
-        :name => [:name],
-        :country => [:country_code, :country],
-        :postal_code => [:postal_code, :zip, :postal],
-        :province => [:province_code, :state_code, :territory_code, :region_code, :province, :state, :territory, :region],
-        :city => [:city, :town],
-        :address1 => [:address1, :address, :street],
-        :address2 => [:address2],
-        :address3 => [:address3],
-        :phone => [:phone, :phone_number],
-        :fax => [:fax, :fax_number],
-        :address_type => [:address_type],
-        :company_name => [:company, :company_name]
-      }
+      return object if object.is_a?(ActiveShipping::Location)
+
       attributes = {}
+
       hash_access = begin
         object[:some_symbol]
         true
       rescue
         false
       end
-      attr_mappings.each do |pair|
+
+      ATTRIBUTE_MAPPINGS.each do |pair|
```

Comment:
While we're here, we can make this a little clearer to read too. What do you think about these changes?

 - Renaming `ATTRIBUTE_MAPPINGS` to `ATTRIBUTE_ALIASES`
 - Change the `each` `pair` parameter to an actual pair of named parameters, `attribute` and `aliases`:
 - Change the `pair[1].each` to be a `detect` since that's what we're really doing anyway with the `break`.

So something like:
```ruby
ATTRIBUTE_ALIASES.each do |attribute, aliases|
  aliases.detect do |alias|
   # blah blah
```
It makes it abundantly clear that we're assigning the aliased attribute's value to whichever aliased attribute what passed in for that attribute.